ReportValidator/validator/iacscanreport_test.go

```diff
@@ -0,0 +1,381 @@
+			severityCounts: map[string]int{
+				"CRITICAL": 0,
+				"MEDIUM":   3,
+				"LOW":      2,
+			},
+			userViolationCount: map[string]int{
+				"CRITICAL": 0,
+				"HIGH":     0,
+				"MEDIUM":   1,
+				"LOW":      1,
+			},
+			expectedFailureCriteriaViolations: map[string]bool{
+				"LOW":      true,
+				"MEDIUM":   true,
+				"HIGH":     false,
+				"CRITICAL": false,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			failureCriteriaViolations := computeViolationState(test.severityCounts, test.userViolationCount)
+
+			if diff := cmp.Diff(test.expectedFailureCriteriaViolations, failureCriteriaViolations); diff != "" {
+				t.Errorf("Expected failureCriteriaViolations (+got, -want): %v", diff)
+			}
+		})
+	}
+}
+
+func TestIsBreachingThreshold(t *testing.T) {
+	tests := []struct {
+		name                      string
+		operator                  string
+		failureCriteriaViolations map[string]bool
+		expectedBool              bool
+		wantErr                   bool
+	}{
+		{
+			name:     "ANDOperator_SeverityNotViolated",
+			operator: "AND",
+			failureCriteriaViolations: map[string]bool{
+				"MEDIUM": true,
+				"HIGH":   false,
+			},
+			expectedBool: false,
+			wantErr:      false,
+		},
+		{
+			name:     "ANDOperator_SeverityViolated",
+			operator: "AND",
+			failureCriteriaViolations: map[string]bool{
+				"MEDIUM": true,
+				"HIGH":   true,
+			},
+			expectedBool: true,
+			wantErr:      false,
+		},
+		{
+			name:     "OROperator_SeverityNotViolated",
+			operator: "OR",
+			failureCriteriaViolations: map[string]bool{
+				"key1": false,
+				"key2": false,
+			},
+			expectedBool: false,
+			wantErr:      false,
+		},
+		{
+			name:     "OROperator_SeverityViolated",
+			operator: "OR",
+			failureCriteriaViolations: map[string]bool{
+				"key1": true,
+				"key2": false,
+			},
+			expectedBool: true,
+			wantErr:      false,
+		},
+		{
+			name:                      "InvalidOperator_Failure",
+			operator:                  "RANDOM",
+			failureCriteriaViolations: map[string]bool{},
+			expectedBool:              true,
+			wantErr:                   true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			isVoilated, err := isBreachingThreshold(test.operator, test.failureCriteriaViolations)
+			if (err != nil) != test.wantErr {
+				t.Errorf("Expected error: %v, got: %v", test.wantErr, err)
+			}
+
+			if test.expectedBool != isVoilated {
+				t.Errorf("Unexpected output want: %v, got: %v", test.expectedBool, isVoilated)
+			}
+		})
+	}
+}
+
+func TestAll(t *testing.T) {
+	tests := []struct {
+		name           string
+		input          map[string]bool
+		expectedOutput bool
+	}{
+		{
+			name:           "AllTrue",
+			input:          map[string]bool{"CRITICAL": true, "HIGH": true, "LOW": true},
+			expectedOutput: true,
+		},
+		{
+			name:           "AllNotTrue",
```

Comment:
Maybe rename to SomeNotTrue? AllNotTrue feels like all of them would be False.

Reply:
Done.